common-moonmage.lic

```diff
@@ -127,15 +127,46 @@ module DRCMM
   # Returns false if you're not holding a moon weapon, or you are but can't wear it.
   # https://elanthipedia.play.net/Shape_Moonblade
   def wear_moon_weapon?
-    moon_weapon_regex = /moon[\w]+/
-    moon_wear_messages = ["telekinetic", "already wearing", "You can't wear", "Wear what"]
+    moon_weapon_regex = /((black|red-hot|blue-white) moon[\w]+)/
+    moon_wear_messages = ["You're already", "You can't wear", "Wear what", "telekinetic"]
     wore_it = false
     if DRC.left_hand =~ moon_weapon_regex
-      wore_it = wore_it || bput("wear #{DRC.left_hand}", *moon_wear_messages) =~ /telekinetic/
+      wore_it = wore_it || DRC.bput("wear #{DRC.left_hand}", *moon_wear_messages) == "telekinetic"
     end
     if DRC.right_hand =~ moon_weapon_regex
-      wore_it = wore_it || bput("wear #{DRC.right_hand}", *moon_wear_messages) =~ /telekinetic/
+      wore_it = wore_it || DRC.bput("wear #{DRC.right_hand}", *moon_wear_messages) == "telekinetic"
     end
     return wore_it
   end
+
+  # Is a moon weapon in your hands?
+  def holding_moon_weapon?
+    moon_weapon_regex = /((black|red-hot|blue-white) moon[\w]+)/
+    return (DRC.left_hand =~ moon_weapon_regex || DRC.right_hand =~ moon_weapon_regex) != nil
+  end
+
+  # Try to hold a moon weapon.
+  # If one isn't already in your hands then will look to see if
+  # one is floating around you and hold it.
+  # If you end up not holding a moon weapon then returns false.
+  def hold_moon_weapon?
+    return true if holding_moon_weapon?
+    moon_weapon_regex = /((black|red-hot|blue-white) moon[\w]+)/
+    moon_floating_regex = /((black|red-hot|blue-white) moon[\w]+) bobs lazily in the air next to you/
+    moon_hold_messages = ["You grab", "You aren't wearing", "Hold hands with whom?"]
+    held_it = false
+    # Look at yourself to see if you're wearing a moon weapon.
+    # This regex specifically checks for the messaging that it's in the air next to you
+    # so that we avoid false positives like wearing a "black moonsilk shirt".
+    look_result = DRC.bput("look #{Char.name}", moon_floating_regex, "You are wearing")
+    # Now try to extract just the moon weapon name from what we saw.
+    scan_result = look_result.scan(moon_weapon_regex)
+    # If a match was found then grab it from the captured groups per the regex.
+    if scan_result.length > 0 && scan_result[0].length > 0
+      moon_weapon = scan_result[0][0]
+      held_it = DRC.bput("hold #{moon_weapon}", *moon_hold_messages) == "You grab"
+    end
+    return held_it
+  end
```

Comment:
Caveat, if you're wearing a black shadowsilk cloak then `LOOK`ing at your character won't see anything so it won't be able know if you're wearing one or not. The only alternative I'm aware of if this use case needs to be supported would be to brute force try to hold each possibility:
- hold moonblade
- hold moonstaff
- hold moonkatar
- ...